contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -1383,112 +1383,112 @@ sp_describe_undeclared_parameters_internal(PG_FUNCTION_ARGS)
 					"WHEN a.atttypmod != -1 THEN "
 						"sys.tsql_type_scale_helper(T2.name, a.atttypmod, false) "
 					"ELSE "
 						"sys.tsql_type_scale_helper(T2.name, t.typtypmod, false) "
-				"END  AS sys.VARCHAR(10) ) + \')\' "
-			"WHEN \'varchar\' THEN "
+				"END  AS sys.VARCHAR(10) ) COLLATE sys.database_default + \')\' COLLATE sys.database_default "
+			"WHEN \'varchar\' COLLATE sys.database_default THEN "
 			"CASE WHEN CASE "
 					"WHEN a.atttypmod != -1 THEN "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, a.atttypmod) "
 					"ELSE "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, t.typtypmod) "
-				"END  = -1 THEN \'varchar(max)\' "
-			"ELSE \'varchar(\' + CAST( CASE "
+				"END  = -1 THEN \'varchar(max)\' COLLATE sys.database_default "
+			"ELSE \'varchar(\' COLLATE sys.database_default + CAST( CASE "
 					"WHEN a.atttypmod != -1 THEN "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, a.atttypmod) "
 					"ELSE "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, t.typtypmod) "
-				"END  AS sys.VARCHAR(10) ) + \')\' "
+				"END  AS sys.VARCHAR(10) ) COLLATE sys.database_default + \')\' COLLATE sys.database_default "
 			"END "
-			"WHEN \'nvarchar\' THEN "
+			"WHEN \'nvarchar\' COLLATE sys.database_default THEN "
 			"CASE WHEN CASE "
 					"WHEN a.atttypmod != -1 THEN "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, a.atttypmod) "
 					"ELSE "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, t.typtypmod) "
-				"END  = -1 THEN \'nvarchar(max)\' "
-			"ELSE \'nvarchar(\' + CAST( CASE "
+				"END  = -1 THEN \'nvarchar(max)\' COLLATE sys.database_default "
+			"ELSE \'nvarchar(\' COLLATE sys.database_default + CAST( CASE "
 					"WHEN a.atttypmod != -1 THEN "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, a.atttypmod) "
 					"ELSE "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, t.typtypmod) "
-				"END /2 AS sys.VARCHAR(10) ) + \')\' "
+				"END /2 AS sys.VARCHAR(10) ) COLLATE sys.database_default + \')\' COLLATE sys.database_default "
 			"END "
-			"WHEN \'varbinary\' THEN "
+			"WHEN \'varbinary\' COLLATE sys.database_default THEN "
 			"CASE WHEN CASE "
 					"WHEN a.atttypmod != -1 THEN "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, a.atttypmod) "
 					"ELSE "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, t.typtypmod) "
-				"END  = -1 THEN \'varbinary(max)\' "
-			"ELSE \'varbinary(\' + CAST( CASE "
+				"END  = -1 THEN \'varbinary(max)\' COLLATE sys.database_default "
+			"ELSE \'varbinary(\' COLLATE sys.database_default + CAST( CASE "
 					"WHEN a.atttypmod != -1 THEN "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, a.atttypmod) "
 					"ELSE "
 						"sys.tsql_type_max_length_helper(T2.name, a.attlen, t.typtypmod) "
-				"END  AS sys.VARCHAR(10) ) + \')\' "
+				"END  AS sys.VARCHAR(10) ) COLLATE sys.database_default + \')\' COLLATE sys.database_default "
```

Comment:
How to make sure we didn't miss anything ?

Reply:
The only feasible way is through tests, we assume authors have already added tests for such procedures, functions, views etc.